packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -810,32 +810,28 @@ export function attach(
   // Patching the console enables DevTools to do a few useful things:
   // * Append component stacks to warnings and error messages
   // * Disable logging during re-renders to inspect hooks (see inspectHooksOfFiber)
-  //
-  // Don't patch in test environments because we don't want to interfere with Jest's own console overrides.
-  if (process.env.NODE_ENV !== 'test') {
```

Comment:
This condition never evaluates to true

Reply:
Ah good catch. We probably want to do `if (!__TEST__)` here instead

Reply:
This breaks these tests:

```
Test Suites: 7 failed, 27 passed, 34 total
Tests:       49 failed, 6 skipped, 383 passed, 438 total
Snapshots:   97 failed, 566 passed, 663 total
```

I haven't looked into why, but it looks like many tests assume that `        registerRendererWithConsole(renderer);` will be called. Do you have any thoughts on whether we want `registerRendererWithConsole` to be called here?

Reply:
Yeah it should be OK to move `registerRendererWithConsole` out of the test because we're not actually overriding the console at all there, just adding some functions for the console override methods to use

Reply:
Looks like wrapping any section in `if (!__TEST__)` breaks at least that many tests. Thoughts on landing this as is?

Reply:
Sure do it :)

Reply:
Yeah, I think when I explore making changes to not use `window.__REACT_DEVTOOLS_GLOBAL_FOO__`-type variables, I should be able to figure it out